magefile.go

```diff
@@ -21,238 +21,55 @@
 package main
 
 import (
-	"errors"
 	"fmt"
-	"io"
-	"io/ioutil"
-	"log"
-	"net/http"
-	"os"
-	"path/filepath"
-	"strings"
-	"text/template"
 
+	// mage:import
+	"github.com/elastic/elastic-agent-libs/dev-tools/mage"
+	"github.com/elastic/elastic-agent-libs/dev-tools/mage/gotool"
 	"github.com/magefile/mage/mg"
-	"github.com/magefile/mage/sh"
-)
-
-const (
-	linterInstallURL             = "https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh"
-	linterInstallFilename        = "./build/intall-golang-ci.sh"
-	linterBinaryFilename         = "./build/golangci-lint"
-	linterVersion                = "v1.44.0"
-	linterConfigFilename         = "./.golangci.yml"
-	linterConfigTemplateFilename = "./dev-tools/templates/.golangci.yml"
-	goVersionFilename            = "./.go-version"
 )
 
 // Aliases are shortcuts to long target names.
 // nolint: deadcode // it's used by `mage`.
 var Aliases = map[string]interface{}{
-	"llc":  Linter.LastChange,
-	"lint": Linter.All,
-}
-
-// Linter contains targets related to linting the Go code
-type Linter mg.Namespace
-
-// UpdateGoVersion updates the linter configuration with the new version of Go.
-func (Linter) UpdateGoVersion() error {
-	goVersionBytes, err := ioutil.ReadFile(goVersionFilename)
-	if err != nil {
-		return fmt.Errorf("failed to read the %q file: %w", goVersionFilename, err)
-	}
-	goVersion := strings.TrimSpace(string(goVersionBytes))
-	log.Printf("The Go version is %s\n", goVersion)
-
-	templateContext := struct{ GoVersion string }{GoVersion: goVersion}
-	template, err := template.ParseFiles(linterConfigTemplateFilename)
-	if err != nil {
-		return fmt.Errorf("failed to read the template file %q: %w", linterConfigTemplateFilename, err)
-	}
```

Comment:
I can't see how this would work now. The version is now an environment variable and there is no mage target that supports it (actually no target at all after this change).

Reply:
When we made these templates we didn't know how the automated Go version bump infrastructure. The process is documented here: https://github.com/elastic/observability-robots/blob/main/docs/teams/ci/bump-versions.md#go-bump

There is a Jenkins job that does the version bump for each repository, it invokes a script to do the version bump for everything in a repository. For this repo, that script is https://github.com/elastic/elastic-agent-autodiscover/blob/5590cbc797a6a94fec29baeacfac035f5f30721c/.ci/bump-go-release-version.sh#L27

The CI will take care of creating a version bump PR once a week when a new version of Go is available. I don't think we need the templates anymore.

Reply:
The environment variable is being read out of the `.go-version` file in CI: https://github.com/elastic/elastic-agent-autodiscover/blob/5590cbc797a6a94fec29baeacfac035f5f30721c/.ci/bump-go-release-version.sh#L27

Reply:
I should also have mentioned that the Jenkins workers that run the automated version bump didn't appear to have Go or Mage installed on them, so the script used just uses sed in place of the mage target that uses the templates.